### DISCUSSION TITLE
Automatically convert components to dynamic imports

### DISCUSSION THREAD

**Discussion by baixiaoyu2997** [nuxt/nuxt#17509] [Ideas]
how about provide a plugin to analyze code coverage, analyze unused or low use components, and convert them to dynamic imports.

I just saw an article about this，but it's a Chinese article

https://mp.weixin.qq.com/s/iQy-Y7yxla3HItKjsNTwng

does vite community have same plugin?

**Comment by pi0** [comment]
Hi @baixiaoyu2997. Nuxt uses dynamic chunks per pages. Components are loaded only when used in those pages and can be optionally lazy loaded with `<Lazy` prefix. Do you have some examples of your idea? A sandbox would be nice to demonstrate.

**Reply by baixiaoyu2997** [reply]
thanks for reply
Sorry, I don't know much about nuxt3 build principle, so I tested the lazy loading component, and found that even if `v-if=false`, the big.js and dayjs in the lazy loading component will still be imported, is this a bug?
I can't confirm if this discussion is meaningful until this issue is resolved
https://codesandbox.io/s/zlzwse

check in  `.output` and `.ouput-lazy` dir

step: 
1. ssr:false
2. in `pages/index.vue`,  switch to lazy input
3. `yarn build`
